item-impl/src/main/java/com/example/auction/item/impl/CompletionStageUtils.java

```diff
@@ -0,0 +1,43 @@
+package com.example.auction.item.impl;
+
+import akka.Done;
+import com.datastax.driver.core.BoundStatement;
+import com.lightbend.lagom.javadsl.persistence.cassandra.CassandraReadSide;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+final class CompletionStageUtils {
+    private CompletionStageUtils() {
+        throw new Error("no instances");
+    }
+
+    static <R, X extends RuntimeException> Function<Optional<R>, R> throwIfEmpty(Supplier<X> exceptionThrower) {
+        return result -> result.orElseThrow(exceptionThrower);
+    }
+
+    static CompletionStage<List<BoundStatement>> completedStatements(BoundStatement... statements) {
+        return CassandraReadSide.completedStatements(Arrays.asList(statements));
+    }
+
+    static <T> Function<T, Done> accept(Consumer<T> f) {
+        return t -> {
+            f.accept(t);
+            return Done.getInstance();
+        };
+    }
+
+    static CompletionStage<Done> doAll(CompletionStage<?>... stages) {
```

Comment:
There's a few reasons.  Firstly, `Void` is not instantiatable, so you must pass `null`.  This causes problems when interacting with other APIs that don't support `null`, such as `Optional.of`, and reactive streams also doesn't allow `null` elements.  Secondly, `Void` doesn't really communicate the intent.  Akka actually introduced two different types `NotUsed` and `Done`.  `NotUsed` is more semantically equivalent to `Void`, it's used where something has to have a parameter type, but that parameter is not used, and so whatever is passed there has no meaning.  In contrast, if something gives you a `Done`, you know whatever it was doing is now done.  A good example of where this distinction matters is if you have `ServiceCall<Source<SomeMessage, ?>, NotUsed>` vs `ServiceCall<Source<SomeMessage, ?>, Done>`. A service call of that shape currently gets implemented using websockets, with the return message being a single message.  If the type is `NotUsed`, Lagom will not wait for a return message, it will immediately return `NotUsed`.  But if it's `Done`, then it treats it as a regular message, and doesn't emit anything until it gets a `Done` message back (which by default is implemented by passing an empty message).